### PR TITLE
Astronomical ephemeris tests

### PR DESCRIPTION
**Pull request by eggrobin**

### CODE REVIEW

File: astronomy/gravity_model.proto.txt [hunk 1]
```diff
@@ -33,8 +33,6 @@ gravity_model {
   }
 
   # Gas giants.
-  # Gravitational parameters from
-  # http://ssd.jpl.nasa.gov/?gravity_fields_op, retrieved 2015-08-03.
```
Comment (pleroy) [inline astronomy/gravity_model.proto.txt]: Why the change?  It's still useful to know where we got the data from.

File: astronomy/solar_system_dynamics_test.cpp [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+    Angle separation_per_orbit;
+    Angle inclination_drift_per_orbit;
+    std::experimental::optional<Angle>
+        longitude_of_ascending_node_drift_per_orbit;
+    std::experimental::optional<Angle>
+        argument_of_periapsis_drift_per_orbit;
+  };
+
+  SolarSystemDynamicsTest() {
+#if 1
```
Comment (pleroy) [inline astronomy/solar_system_dynamics_test.cpp]: Why?

File: astronomy/solar_system_dynamics_test.cpp [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+    google::LogToStderr();
+#endif
+    for (int primary = 0; primary <= SolarSystemFactory::LastBody; ++primary) {
+      for (int i = 1; i <= SolarSystemFactory::LastBody; ++i) {
```
Comment (pleroy) [inline astronomy/solar_system_dynamics_test.cpp]: Might as well start at 0 here too.

File: astronomy/solar_system_dynamics_test.cpp [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+                         GravitationalParameter> expected_subsystem_barycentre;
+
+    actual_subsystem_barycentre.Add(
+        ephemeris.trajectory(body)->
+            EvaluateDegreesOfFreedom(epoch, /*hint=*/nullptr),
+        body->gravitational_parameter());
+    expected_subsystem_barycentre.Add(expected_system.initial_state(name),
+                                      body->gravitational_parameter());
+    for (int const moon_index : bodies_orbiting_[index]) {
+      auto const moon_name = SolarSystemFactory::name(moon_index);
```
Comment (pleroy) [inline astronomy/solar_system_dynamics_test.cpp]: auto -> string?

File: astronomy/solar_system_dynamics_test.cpp [hunk 5]
```diff
@@ -0,0 +1,352 @@
...
+    enum LocalFrameTag { tag };
+    using ParentEquator = Frame<LocalFrameTag, tag, /*frame_is_inertial=*/true>;
+    auto const z = Bivector<double, ICRFJ2000Equator>({0, 0, 1});
+    std::experimental::optional<Rotation<ICRFJ2000Equator, ParentEquator>>
+        rotation;
+
+    if (SolarSystemFactory::parent(index) == SolarSystemFactory::Sun) {
+      Bivector<AngularMomentum, ICRFJ2000Equator>
+          solar_system_angular_momentum;
+      for (int i = 1; i <= SolarSystemFactory::LastBody; ++i) {
```
Comment (pleroy) [inline astronomy/solar_system_dynamics_test.cpp]: 1 -> kSun + 1

File: astronomy/solar_system_dynamics_test.cpp [hunk 6]
```diff
@@ -0,0 +1,352 @@
...
+                               parent->angular_velocity()),
+                  AllOf(Gt(5.9 * Degree), Lt(6.0 * Degree)));
+      auto const declination_of_invariable_plane =
+          AngleBetween(solar_system_angular_momentum, z);
+      rotation =
+          Rotation<ICRFJ2000Equator, ParentEquator>(
+              declination_of_invariable_plane,
+              normal);
+    } else {
+      auto const ω = parent->angular_velocity();
```
Comment (pleroy) [inline astronomy/solar_system_dynamics_test.cpp]: In general this code seems to overuse `auto` a bit.  I would rather see a real type in many of these declarations.

File: astronomy/solar_system_dynamics_test.cpp [hunk 7]
```diff
@@ -0,0 +1,352 @@
...
+
+    OrbitError result;
+    result.separation_per_orbit =
+        AngleBetween(actual_dof.position() - actual_parent_dof.position(),
+                     expected_dof.position() - expected_parent_dof.position()) /
+        orbits;
+    result.inclination_drift_per_orbit =
+        AbsoluteError(expected_elements.inclination,
+                      actual_elements.inclination) / orbits;
+    if (actual_elements.inclination > 0.1 * Degree) {
```
Comment (pleroy) [inline astronomy/solar_system_dynamics_test.cpp]: Could we use a namespace-level constant for the tolerance here?

File: astronomy/solar_system_dynamics_test.cpp [hunk 8]
```diff
@@ -0,0 +1,352 @@
...
+      result.longitude_of_ascending_node_drift_per_orbit =
+          AbsoluteError(expected_elements.longitude_of_ascending_node,
+                        actual_elements.longitude_of_ascending_node) / orbits;
+
+      if (actual_elements.eccentricity > 0.1) {
```
Comment (pleroy) [inline astronomy/solar_system_dynamics_test.cpp]: And here too.

File: astronomy/solar_system_dynamics_test.cpp [hunk 9]
```diff
@@ -0,0 +1,352 @@
...
+      solar_system_at_j2000.MakeEphemeris(
+          /*fitting_tolerance=*/5 * Milli(Metre),
+          Ephemeris<ICRFJ2000Equator>::FixedStepParameters(
+              integrators::BlanesMoan2002SRKN14A<Position<ICRFJ2000Equator>>(),
+              /*step=*/45 * Minute));
+  ephemeris->Prolong(ten_years_later.epoch());
+
+  for (int const planet_or_minor_planet :
+       bodies_orbiting_[SolarSystemFactory::Sun]) {
+    LOG(INFO) << "=== " << SolarSystemFactory::name(planet_or_minor_planet);
```
Comment (pleroy) [inline astronomy/solar_system_dynamics_test.cpp]: I'd be fine with logging to ERROR in all these traces.

File: astronomy/gravity_model.proto.txt (hunk 1 above)
Comment (eggrobin) [inline astronomy/gravity_model.proto.txt]: Because we don't get it from there anymore.

File: astronomy/solar_system_dynamics_test.cpp (hunk 2 above)
Comment (eggrobin) [inline astronomy/solar_system_dynamics_test.cpp]: Removed the #if.

File: astronomy/solar_system_dynamics_test.cpp (hunk 3 above)
Comment (eggrobin) [inline astronomy/solar_system_dynamics_test.cpp]: `parent(i)` would `LOG(FATAL)`.

File: astronomy/solar_system_dynamics_test.cpp (hunk 4 above)
Comment (eggrobin) [inline astronomy/solar_system_dynamics_test.cpp]: Done.

File: astronomy/solar_system_dynamics_test.cpp (hunk 5 above)
Comment (eggrobin) [inline astronomy/solar_system_dynamics_test.cpp]: doen.

File: astronomy/solar_system_dynamics_test.cpp (hunk 6 above)
Comment (eggrobin) [inline astronomy/solar_system_dynamics_test.cpp]: added a couple of types.
